doc/nrf/gs_modifying.rst

```diff
@@ -14,27 +14,53 @@ After programming and testing an application, you probably want to make some mod
 Adding files and changing compiler settings
 *******************************************
 
-All files that your application uses must be specified in the :file:`CMakeLists.txt` file.
-By default, most samples include only the main application file :file:`src/main.c`.
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, whose build system is based on `CMake`_.
+For more information, see :ref:`zephyr:application` in the Zephyr documentation.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+In the |NCS|, the application is a CMake project.
+As such, the application controls the configuration and build process of both itself, Zephyr, and sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main CMake project file and the source of the build process configuration.
+
+Zephyr creates a CMake component that must be imported by the application into its :file:`CMakeLists.txt` file.
```

Comment:
> Zephyr creates a CMake component

Considering this is about the build system then this sounds to me like something being created during the build process which is not the case.

How about:
```suggestion
Zephyr provides a CMake package that must be loaded by the application into its :file:`CMakeLists.txt` file.
```

*Note*: CMake provides two types, `packages` and `modules` so as a reader knowing CMake, I would be in doubt which of those you mean when referring to `component`:
https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html
https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html
https://cmake.org/cmake/help/latest/command/find_package.html#command:find_package

Ref:
> **find_package**
> Find a package (usually provided by something external to the project), and load its package-specific details.

Reply:
I meant package.